doc/source/operations/gpu-in-openstack.rst

```diff
@@ -2,6 +2,132 @@
 =============================
 
+PCI Passthrough
+###############
+
+Prerequisite - BIOS Configuration
+---------------------------------
+
+On an Intel system:
+
+* Enable ``VT-x`` in the BIOS for virtualisation support.
+* Enable ``VT-d`` in the BIOS for IOMMU support.
+
+On an AMD system:
+
+* Enable ``AMD-v`` in the BIOS for virtualisation support.
+* Enable ``AMD-Vi`` (also just called ``IOMMU`` on older hardware) in the BIOS
+  for IOMMU support.
+
+It may be possible to configure passthrough without these settings, though
+stability or performance may be affected.
+
+Host and Service Configuration
+------------------------------
+
+PCI passthrough GPU variables can be found in the
+``etc/kayobe/stackhpc-compute.yml`` file.
+
+The ``gpu_group_map`` is a dictionary mapping inventory groups to GPU types.
+This is used to determine which GPU types each compute node should pass through
+to OpenStack. The keys are group names, the values are a list of GPU types.
+
+Possible GPU types are defined in the ``stackhpc_gpu_data`` dictionary. It
+contains data for many common GPUs. If you have a GPU that is not included,
+extend the dictionary following the same pattern.
+
+The ``resource_name`` is the name that will be used in the flavor extra specs.
+These can be overridden e.g. ``a100_80_resource_name: "big_gpu"``.
+
+Example configuration for three groups containing A100s, V100s, and both:
+
+.. code-block:: yaml
+   :caption: $KAYOBE_CONFIG_PATH/stackhpc-compute.yml
+
+    gpu_group_map:
+      compute_a100:
+        - a100_80
+      compute_v100:
+        - v100_32
+      compute_multi_gpu:
+        - a100_80
+        - v100_32
+
+All groups in the ``gpu_group_map`` must also be added to
+``kolla_overcloud_inventory_top_level_group_map`` in ``etc/kayobe/kolla.yml``.
+Always include the Kayobe defaults unless you know what you are doing.
+
+When ``gpu_group_map`` is populated, the ``pci-passthrough.yml`` playbook will
+be added as a pre-hook to ``kayobe overcloud host configure``. Either run host
+configuration or trigger the playbook manually:
+
+.. code-block:: console
+
+    kayobe overcloud host configure --limit compute_a100,compute_v100,compute_multi_gpu
+    # OR
+    kayobe playbook run --playbook $KAYOBE_CONFIG_PATH/ansible/pci-passthrough.yml --limit compute_a100,compute_v100,compute_multi_gpu
+
+The playbook will apply the necessary configuraion and reboot the hosts if
+required.
+
+Once host configuration is complete, deploy the OpenStack services:
+.. code-block:: console
+
+    kayobe overcloud service deploy -kt nova --kolla-limit compute_a100,compute_v100,compute_multi_gpu
+
+Create a flavor
+---------------
+
+For example, to request two of the GPUs with alias **v100_32**
+
+.. code-block:: text
+
+   openstack flavor set m1.medium-gpu --property "pci_passthrough:alias"="v100_32:2"
+
+This can be also defined in the openstack-config repository
+
+add extra_specs to flavor in etc/openstack-config/openstack-config.yml:
```

Comment:
```suggestion
This can be also defined in the openstack-config repository.

Add extra_specs to flavor in etc/openstack-config/openstack-config.yml:
```